lib/rules/no-reserved-component-names.js

```diff
@@ -109,6 +88,26 @@ module.exports = {
       options.disallowVueBuiltInComponents === true
     const disallowVue3BuiltInComponents =
       options.disallowVue3BuiltInComponents === true
+    const htmlElementCaseSensitive = options.htmlElementCaseSensitive === true
+
+    const RESERVED_NAMES_IN_HTML = new Set([
+      ...htmlElements,
+      ...(htmlElementCaseSensitive ? [] : htmlElements.map(casing.capitalize))
+    ])
+    const RESERVED_NAMES_IN_OTHERS = new Set([
+      ...deprecatedHtmlElements,
+      ...(htmlElementCaseSensitive
+        ? []
+        : deprecatedHtmlElements.map(casing.capitalize)),
+      ...kebabCaseElements,
+      ...(htmlElementCaseSensitive
+        ? []
+        : kebabCaseElements.map(casing.pascalCase)),
+      ...svgElements,
+      ...(htmlElementCaseSensitive
+        ? []
+        : svgElements.filter(isLowercase).map(casing.capitalize))
+    ])
```

Comment:
I think imperative rather than functional style is more readable here:

```js
// helper function in top scope
/**
 * @template T
 * @param {Set<T>} set
 * @param {Iterable<T>} iterable
 */
function addAll(set, iterable) {
  for (const element of iterable) {
    set.add(element)
  }
}
```

```js
    const htmlElementCaseSensitive = options.htmlElementCaseSensitive === true

    const RESERVED_NAMES_IN_HTML = new Set(htmlElements)
    const RESERVED_NAMES_IN_OTHERS = new Set([
      ...deprecatedHtmlElements,
      ...kebabCaseElements,
      ...svgElements,
    ])
    
    if (!htmlElementCaseSensitive) {
      addAll(RESERVED_NAMES_IN_HTML, htmlElements.map(casing.capitalize))
      addAll(RESERVED_NAMES_IN_OTHERS, [
        ...deprecatedHtmlElements.map(casing.capitalize))
        ...kebabCaseElements.map(casing.pascalCase))
        ...svgElements.filter(isLowercase).map(casing.capitalize))
      ])
    }
```